pkg/module_manager/global_hook.go

```diff
@@ -190,10 +212,27 @@ func (h *GlobalHook) Run(bindingType BindingType, context []BindingContext, logL
 		if err != nil {
 			return fmt.Errorf("global hook '%s': dynamic global values update error: %s", h.Name, err)
 		}
+
 		// MemoryValuesPatch from global hook can contains patches for *Enabled keys
 		// and no patches for 'global' section — valuesPatchResult will be nil in this case.
 		if valuesPatchResult != nil && valuesPatchResult.ValuesChanged {
-			h.moduleManager.globalDynamicValuesPatches = utils.AppendValuesPatch(h.moduleManager.globalDynamicValuesPatches, valuesPatchResult.ValuesPatch)
+			log.Debugf("Global hook '%s': validate global values before update", h.Name)
+			validationErr := validation.ValidateGlobalValues(valuesPatchResult.Values)
```

Comment:
This is the only place where we merge values after validation. I got the idea, but maybe a little comment will help others in the future.

Reply:
Values are merged by handleGlobalValuesPatch and stored in valuesPatchResult.Values. I'll add comments here and in module_hook.go.